k3s/k3s-init.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+# Check k3s installation
+k3s_path=$(which k3s)
+if [ -x "$k3s_path" ]; then
+    echo "k3s is already installed"
+    exit 0
+fi
+
+# Install k3s
+$ curl -sfL https://get.k3s.io | K3S_KUBECONFIG_MODE="644" sh -s -
+
+K3S_KUBECONFIG=/etc/rancher/k3s/k3s.yaml
+
+# Check if kubectl is installed
+echo "Checking if kubectl is in PATH..."
+kubectl_path=$(which kubectl)
+if [ -x "$kubectl_path" ]; then
+    echo "kubectl is installed, merging kubeconfig files..."
+    
+    # Backup the existing kubeconfig file if it exists
+    if [ -f ~/.kube/config ]; then
+        KUBECONFIG_BACKUP=~/.kube/config.backup
+        cp ~/.kube/config $KUBECONFIG_BACKUP
```

Comment:
This cp should use the -p option to preserve the existing file perms and attributes.
```suggestion
        cp -p ~/.kube/config $KUBECONFIG_BACKUP
```